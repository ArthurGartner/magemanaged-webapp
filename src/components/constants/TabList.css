.tab-list {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

.page-tab {
  height: 30px;
  margin: .5em 0;
  margin: 10px 0;
}

.card-tab {
  margin-left: 10px;
}
/*# sourceMappingURL=TabList.css.map */