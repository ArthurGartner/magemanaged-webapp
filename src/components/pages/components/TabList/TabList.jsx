import React from "react";
import { Tab } from "./components/Tab";
import "./TabList.css";

export const TabList = (props) => {
  return (
    <div className={`${props.style}`}>
      <ul className="tab-list">
        {props.items.map((tabName) => (
          <li key={tabName}>
            <Tab
              name={tabName}
              click={props.click}
              selected={props.selected}
              color={props.color}
            />
          </li>
        ))}
      </ul>
    </div>
  );
};
