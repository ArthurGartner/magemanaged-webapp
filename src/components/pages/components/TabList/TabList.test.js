import React from "react";
import { render, cleanup } from "@testing-library/react";
import { TabList } from "./TabList";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

//Must have atleast one tab to map
describe("TabList", () => {
  it("renders without crashing", () => {
    render(<TabList items={["Test Title"]} />);
  });
});

it("matches snapshot", () => {
  const tree = renderer.create(<TabList items={["Test Title"]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
