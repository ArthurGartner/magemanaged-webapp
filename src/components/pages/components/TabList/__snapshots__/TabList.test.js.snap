// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="undefined"
>
  <ul
    className="tab-list"
  >
    <li>
      <div
        className="tab-container"
        data-testid="tab-test"
      >
        <button
          className="tab-item undefined"
          onClick={[Function]}
        >
          Test Title
        </button>
      </div>
    </li>
  </ul>
</div>
`;
